layer/migration.go

```diff
@@ -178,6 +173,18 @@ func (ls *layerStore) RegisterByGraphID(graphID string, parent ChainID, tarDataF
 		referenceCount: 1,
 		layerStore:     ls,
 		references:     map[Layer]struct{}{},
+		diffID:         diffID,
+		size:           size,
+		chainID:        createChainIDFromParent(parent, diffID),
+	}
+
+	ls.layerL.Lock()
```

Comment:
Why move this up here? It was originally done inside the transaction after data files has been written to decrease the amount of time this lock needs to be held.